### PR TITLE
Update-2022-12-28

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Update Frontend](https://trello.com/c/gXCYbxlm/1158-update-frontend)

File: projects/core/src/lib/datasets/add-data/add-data.component.html [hunk 1]
```diff
@@ -2,11 +2,11 @@
```
Comment (jdroenner) [inline projects/core/src/lib/datasets/add-data/add-data.component.html:2]: Icon button not centered 

image

File: projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss [hunk 2]
```diff
@@ -52,10 +52,6 @@ mat-progress-bar {
     padding: 0 !important;
 }
```
Comment (jdroenner) [inline projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss:54]: Layer name and layer menu font size not equal
image

File: projects/core/src/lib/plots/plot-list-entry/plot-list-entry.component.scss [hunk 3]
```diff
@@ -3,7 +3,7 @@ img {
     height: auto;
 }
```
Comment (jdroenner) [inline projects/core/src/lib/plots/plot-list-entry/plot-list-entry.component.scss:5]: icons below plots are very small and not centered

image

Comment (jdroenner) [inline projects/core/src/lib/plots/plot-list-entry/plot-list-entry.component.scss:5]: strange green color in "Temporal Raster Mean Plot"  component.

image

File: projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <geoengine-sidenav-header>Calculate Expression on Raster</geoengine-sidenav-header>
```
Comment (jdroenner) [inline projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:2]: "-" "+" icons in operator input not centered (on hover).

image

**Review by jdroenner — CHANGES_REQUESTED** [review]
most issues are font size and icon related

File: projects/gis/src/app/main/main.component.html [hunk 5]
```diff
@@ -28,17 +28,17 @@
...
-    <geoengine-navigation [buttons]="navigationButtons | async | valueDefault: []"></geoengine-navigation>
+    <geoengine-navigation [buttons]="navigationButtons | async | valueDefault : []"></geoengine-navigation>
 
     <geoengine-layer-list
         class="mat-elevation-z4"
         [height]="middleContainerHeight$ | async | numberSanitizer"
-        [addAFirstLayerSidenavConfig]="addDataConfig | async | valueDefault: undefined"
+        [addAFirstLayerSidenavConfig]="addDataConfig | async | valueDefault : undefined"
     >
     </geoengine-layer-list>
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/gis/src/app/main/main.component.html:39]: Yes, MDC has slightly different sizes and margins.

File: projects/gis/src/app/main/main.component.html [hunk 6]
```diff
@@ -28,17 +28,17 @@
         <geoengine-sidenav-container></geoengine-sidenav-container>
     </mat-sidenav>
 
-    <geoengine-navigation [buttons]="navigationButtons | async | valueDefault: []"></geoengine-navigation>
+    <geoengine-navigation [buttons]="navigationButtons | async | valueDefault : []"></geoengine-navigation>
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/gis/src/app/main/main.component.html:32]: [494b00c](https://github.com/geo-engine/geoengine-ui/pull/555/commits/494b00c47d1fcfeb217136cca8cc53eb53893be5)

File: projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss:54]: image

Was already there… but I removed it:

[e805710](https://github.com/geo-engine/geoengine-ui/pull/555/commits/e805710b1a96488b9d849ce374d7035d9b001762)

File: projects/core/src/lib/datasets/add-data/add-data.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/datasets/add-data/add-data.component.html:2]: [d8c4ef7](https://github.com/geo-engine/geoengine-ui/pull/555/commits/d8c4ef703953de9e0e4a52c7d92ec5e9b9775236)

File: projects/core/src/lib/datasets/upload/upload.component.html [hunk 7]
```diff
@@ -159,35 +159,35 @@
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/datasets/upload/upload.component.html:159]: [b54520d](https://github.com/geo-engine/geoengine-ui/pull/555/commits/b54520d8a80f07a5c3cb6f9b104a805cf7604064)

File: projects/core/src/lib/datasets/add-data/add-data.component.html (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/datasets/add-data/add-data.component.html:2]: [bc3f138](https://github.com/geo-engine/geoengine-ui/pull/555/commits/bc3f1384584cc0cdfc93122f587c27d7a6325e60)

File: projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss:54]: I guess its more that the layer menu manually sets a random font size that is not n'sync with the fon size that is used by a component. I changed it… but we should rethink this for future refactorings.

[89ff6e5](https://github.com/geo-engine/geoengine-ui/pull/555/commits/89ff6e588663cc51144cc98fafb5711ae0330b65)

File: projects/core/src/lib/plots/plot-list-entry/plot-list-entry.component.scss (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/plots/plot-list-entry/plot-list-entry.component.scss:5]: They are now default icon buttons:
[8d44fa4](https://github.com/geo-engine/geoengine-ui/pull/555/commits/8d44fa47bbc0e4a1d1464b63ebd6052394ac0c8a)

Comment (ChristianBeilschmidt) [reply projects/core/src/lib/plots/plot-list-entry/plot-list-entry.component.scss:5]: Do you mean the checkbox?
It's part of the theme, but I changed it to primary :shrug: .
[f222a95](https://github.com/geo-engine/geoengine-ui/pull/555/commits/f222a95544c88185ad7c204445d73f2020b38adf)

File: projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:2]: image

For me, it looks like if they are.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -24,7 +24,8 @@ jobs:
                     node-version: ${{ matrix.node-version }}
                     # cache: 'npm' # can be used only with a lockfile
             #            -   run: npm ci    # use instead of install if we choose to commit the `package-lock.json`
-            -   run: npm install
+            # TODO: remove `--force` when @angular/flex-layout is removed
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply .github/workflows/ci.yml:27]: so we have to remove flex-layout or will there be a new one?

File: projects/gis/src/app/main/main.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (jdroenner) [reply projects/gis/src/app/main/main.component.html:39]: i'm fine to keep the MDC sizes but the sizes in the menu tabs and in "+" do not match, right?

File: projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html (hunk 4 above)
Comment (jdroenner) [reply projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:2]: Peek 2023-01-09 10-20

File: .github/workflows/ci.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply .github/workflows/ci.yml:27]: We have to remove it.

File: projects/gis/src/app/main/main.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/gis/src/app/main/main.component.html:39]: No, they do not necessarily match.

File: projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:2]: I now use icon buttons, they seem appropriate. I guess other buttons are designed to have at least some text next to the icon.

[6081ec4](https://github.com/geo-engine/geoengine-ui/pull/555/commits/6081ec4f8839694c35f576e999239086f1826c1c)

File: projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.scss:54]: [6081ec4](https://github.com/geo-engine/geoengine-ui/pull/555/commits/6081ec4f8839694c35f576e999239086f1826c1c)

File: .github/workflows/ci.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (jdroenner) [reply .github/workflows/ci.yml:27]: -_-

File: projects/gis/src/app/main/main.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (jdroenner) [reply projects/gis/src/app/main/main.component.html:39]: do we want that?

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**